src/ray/gcs/gcs_server/gcs_job_manager.h

```diff
@@ -88,7 +96,24 @@ class GcsJobManager : public rpc::JobInfoHandler {
 
   void WriteDriverJobExportEvent(rpc::JobTableData job_data) const;
 
+  /// Record metrics.
+  /// For job manager, (1) running jobs count gauge and (2) new finished jobs (whether
+  /// succeed or fail) will be reported periodically.
+  void RecordMetrics();
+
  private:
+  void ClearJobInfos(const rpc::JobTableData &job_data);
+
+  void MarkJobAsFinished(rpc::JobTableData job_table_data,
+                         std::function<void(Status)> done_callback);
+
+  // Job ids, which are running.
+  absl::flat_hash_set<JobID> running_job_ids_;
+
+  // Used to report metrics, which indicates the number of jobs finished between two
+  // metrics reporting events.
+  int32_t new_finished_jobs_ = 0;
```

Comment:
what about: 

  // Running Job IDs, used to report metrics.
  absl::flat_hash_set<JobID> running_job_ids_;

// Number of finished jobs since start of this GCS Server, used to report metrics.
int64_t finished_jobs_count_ = 0;

Reply:
Reason you don't need `new`: Prometheus Counter `rate()` function auto handles counter resets.

Reply:
Sounds good! Updated.